### PR TITLE
Image generation cancellation

### PR DESCRIPTION
**Pull request by sanjeev-cs**

### CODE REVIEW

File: backend/routes/image_generation/image_generation_api.js [hunk 1]
```diff
@@ -297,7 +299,13 @@ router.get("/poll/:jobId",
       return res.status(400).json({ success: false, error: "Invalid job Id." });
     }
 
-    try {
+    try {  
+      
+      if (errorCount >= maxErrorRetries) {
```
Comment (ConnorBP) [inline backend/routes/image_generation/image_generation_api.js:304]: error count does not exist

File: backend/routes/image_generation/image_generation_api.js [hunk 2]
```diff
@@ -2,13 +2,14 @@ import express from "express";
...
-import { saveImageJobData, getImageJobData, updateImageJobStatus, connectToDB, doesJobIdExist } from "../database/dbService.js";
+import { saveImageJobData, getImageJobData, updateImageJobStatus, connectToDB, doesJobIdExist, cancelImageJob} from "../database/dbService.js";
 import { authenticate } from "../auth/authentication.js";
 
 const router = express.Router();
 const baseUrl = "https://stablehorde.net/api/v2";
 const apiKey = process.env.API_KEY;
-
+const maxErrorRetries = process.env.MAX_ERROR_RETRIES;
+let errorRetries = 0;
```
Comment (ConnorBP) [inline backend/routes/image_generation/image_generation_api.js:12]: error retries will never persist between api calls

File: backend/routes/image_generation/image_generation_api.js [hunk 3]
```diff
@@ -2,13 +2,14 @@ import express from "express";
...
 import { query, check, validationResult } from "express-validator";
-import { saveImageJobData, getImageJobData, updateImageJobStatus, connectToDB, doesJobIdExist } from "../database/dbService.js";
+import { saveImageJobData, getImageJobData, updateImageJobStatus, connectToDB, doesJobIdExist, cancelImageJob} from "../database/dbService.js";
 import { authenticate } from "../auth/authentication.js";
 
 const router = express.Router();
 const baseUrl = "https://stablehorde.net/api/v2";
 const apiKey = process.env.API_KEY;
-
+const maxErrorRetries = process.env.MAX_ERROR_RETRIES;
```
Comment (ConnorBP) [inline backend/routes/image_generation/image_generation_api.js:11]: max tries logic is supposed to be on front end

**Review by ConnorBP — CHANGES_REQUESTED** [review]
The max tries logic should be on the front end in the JobWatcherProvider, not in the server backend

File: backend/routes/image_generation/image_generation_api.js [hunk 4]
```diff
@@ -259,6 +259,7 @@ router.post("/generate", [
 
     res.status(202).json({ success: true, jobId });
   } catch (e) {
+    errorRetries += 1;
```
Comment (ConnorBP) [inline backend/routes/image_generation/image_generation_api.js:262]: This is still broken. errorRetries is not defined, nor would it's state persist anyways. Disable the max retries logic on backend please

**Review by ConnorBP — CHANGES_REQUESTED** [review]

File: backend/routes/image_generation/image_generation_api.js (hunk 1 above)
Comment (sanjeev-cs) [reply backend/routes/image_generation/image_generation_api.js]: Removed.

File: backend/routes/image_generation/image_generation_api.js (hunk 2 above)
Comment (sanjeev-cs) [reply backend/routes/image_generation/image_generation_api.js]: Removed.

File: backend/routes/image_generation/image_generation_api.js (hunk 3 above)
Comment (sanjeev-cs) [reply backend/routes/image_generation/image_generation_api.js]: Removed.

File: backend/routes/image_generation/image_generation_api.js (hunk 4 above)
Comment (sanjeev-cs) [reply backend/routes/image_generation/image_generation_api.js]: Thanks for letting me know. 👍

File: frontend/src/css/ProgressToastOverlay.css [hunk 5]
```diff
@@ -23,4 +23,14 @@
...
+}
+
+/* Allow pointer events specifically for the Cancel button */
+.clear-job-btn, confirm-btn, cancel-btn {
+    pointer-events: auto;
+}
+
+/* Allow pointer events on the popup overlay and its contents */
+.clear-job-popup-overlay, .popup-content, .popup-actions, .confirm-btn, .cancel-btn {
+    pointer-events: auto;
```
Comment (ConnorBP) [inline frontend/src/css/ProgressToastOverlay.css:35]: Add !important and something stronger than auto

File: frontend/src/css/ProgressToastOverlay.css [hunk 6]
```diff
@@ -23,4 +23,14 @@
 
 .progress-toast-overlay, .progress-toast-overlay >* {
     pointer-events: none;
+}
+
+/* Allow pointer events specifically for the Cancel button */
+.clear-job-btn, confirm-btn, cancel-btn {
+    pointer-events: auto;
```
Comment (ConnorBP) [inline frontend/src/css/ProgressToastOverlay.css:30]: !important and not auto here also

**Review by ConnorBP — APPROVED** [review]
woot woot

**Pull request merged**